addons/event_sale/models/sale_order.py

```diff
@@ -67,3 +69,28 @@ def _get_product_catalog_domain(self):
         """
         domain = super()._get_product_catalog_domain()
         return expression.AND([domain, [('service_tracking', '!=', 'event')]])
+
+    def _notify_get_recipients_groups(self, message, model_description, msg_vals=None):
+        groups = super()._notify_get_recipients_groups(message, model_description, msg_vals)
+        if not self or self.state != 'sale' or not self.order_line.registration_ids:
+            return groups
+
+        try:
+            customer_portal_group = next((group for group in groups if group[0] == 'portal_customer'), None)
+        except StopIteration:
+            return groups
+
+        if customer_portal_group[2]['has_button_access']:
+            actions_opt = customer_portal_group[2].setdefault('actions', [])
+            has_single_event = len(self.order_line.event_id) == 1
+            registrations = self.order_line.registration_ids
+            for event in self.order_line.event_id:
+                event_registrations_ids = registrations.filtered(lambda reg: reg.event_id == event).ids
+                actions_opt.append({
```

Comment:
I'm a bit confused between the template and the excalidraw :smile: 
This will make one button per event, but just as links, not as 'buttons', right? 


```xml
<td t-if="has_button_access" t-att-style="'border-radius: 3px; text-align: center; background: ' + (company.email_secondary_color or '#875A7B') + ';'">
    <a t-att-href="button_access['url']" style="font-size: 12px; color: #FFFFFF; display: block; padding: 8px 12px 11px; text-decoration: none !important; font-weight: 400;">
        <t t-out="button_access['title']"/>
    </a>
</td>
<td t-if="has_button_access">&amp;nbsp;&amp;nbsp;</td>

<td t-if="actions">
    <t t-foreach="actions" t-as="action">
        <a t-att-href="action['url']" t-att-style="'font-size: 12px; color: ' + (company.email_secondary_color or '#875A7B')+ '; text-decoration:none !important;'">
            <t t-out="action['title']"/>
        </a>
        &amp;nbsp;&amp;nbsp;
    </t>
</td>
```

Reply:
Yes indeed. I talked a bit about it with Luc and that's the default design we already used in CRM so it's fine. My only point I'm not sure yet is if I display these links only when there is the button_access like now or everytime

Reply:
Yes, would keep current display of actions... that way it is standard :)